src/LoginFormContainer.jsx

```diff
@@ -0,0 +1,41 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  requestLogin, resetAllForm, setAccessToken, setForm,
+} from './actions';
+import LoginForm from './LoginForm';
+
+export default function LoginFormContainer() {
+  const dispatch = useDispatch();
+
+  const { email, password } = useSelector((state) => state.form);
+
+  const accessToken = useSelector((state) => state.accessToken);
+  const isLoggedIn = (accessToken !== null);
+
+  function handleChange({ name, value }) {
+    dispatch(setForm({ name, value }));
+  }
+
+  function handleSubmitByLoginState(loggedIn) {
+    if (!loggedIn) {
+      return () => dispatch(requestLogin());
+    }
+
+    return () => {
+      localStorage.setItem('accessToken', null);
+
+      dispatch(setAccessToken(null));
+      dispatch(resetAllForm());
+    };
+  }
+
+  return (
+    <LoginForm
+      form={{ email, password }}
+      handleChange={handleChange}
+      handleSubmit={handleSubmitByLoginState(isLoggedIn)}
+      isLoggedIn={isLoggedIn}
```

Comment:
```suggestion
      form={form}
      handleChange={handleChange}
      handleSubmit={handleSubmitByLoginState(isLoggedIn)}
      isLoggedIn={isLoggedIn}
```

다른 props 들과 마찬가지로 추상화 레벨을 맞춰줘도 좋을 것 같아요.
따로 destructuring 해주는 의도가 있나요?

Reply:
제가 `TextFields`의  내부 구현을 착각하고 있었네요. `TextField`에 값을 넘기기 전에 필요한 것만 뽑을 필요가 있다고 생각했는데 다시 보니가 불필요하군요. 수정해줬습니다!